netket/stats/mc_stats.py

```diff
@@ -155,101 +203,34 @@ def statistics(data, batch_size=32):
              dict sintax (e.g. res['mean']), one can also access them directly with the dot operator
              (e.g. res.mean).
     """
-    return _statistics(data, batch_size)
+    return _statistics(data, block_size, precompute)
 
 
-@partial(jax.jit, static_argnums=1)
-def _statistics(data, batch_size):
+@partial(jax.jit, static_argnums=(1, 2))
+def _statistics(data, block_size, precompute):
```

Comment:
Can we make this a static method of `Stats`? Something like
```python
stats = Stats.from_data(data, block_size=...)
```
is easier to understand than
```python
stats = nk.stats.mc_stats._statistics(...)
```

Do we really need the `precompute` option? Or, if yes, could we add a sensible default?

Reply:
If we go down the 2 classes route like I did in my last commit, I think that is not possible...

Reply:
As for precompute: it's a performance optimisation.
when I merge different Stats objects I don't need to do MPI reduction operations. I only need to do that at the end, in order to compute the sample mean, variance Rhat... etc.

Therefore I made them cached properties, so they are computed once, only when needed.

However, being written in jax, this slows down a lot the resulting code if they are called one-by-one instead of being computed in the same jax kernel.
If we don't precompute them, the logger will access them 1 by 1 when serializing, slowing down the code.

However, in light of mini batches, we don't want to precompute them all the time.
so I'm leaving the possibility open to decide whever to precompute them or not.